### PR TITLE
add lateral navigation for faceted and layered charts

### LINKED ISSUE #44: implement lateral navigation
We define lateral navigation for faceted charts as being able to jump to the same location in a different sub-branch of the tree. Lateral navigation was implemented in the original research prototypes, but is currently missing in Olli.

For an example of lateral navigation, go to this prototype and go into the y axis for one of the charts. Then use shift + left or shift + right to jump over to the y axis but for the prev/next chart. This also works on intervals.
https://mitvis.github.io/access-interaction/prototypes/jzong/multiview/

Implementation for that prototype: https://github.com/mitvis/access-interaction/blob/958e8d9d9ee426081ac1f478613a70e86ebac23b/prototypes/jzong/multiview/src/MultiViewVis.js#L330C27-L342

In TreeItem.ts, we should add event handling for shift+left and shift+right, and add logic that:
- checks if we're on a faceted chart
- if so, finds the equivalent node in the prev/next facet
- sets focus to that node

### PR DESCRIPTION
**Pull request by ezralee**
Adds lateral navigation for faceted and layered charts. Resolves #44

### CODE REVIEW

**Comment by ezralee** [comment]
Slight weirdness in the layered example since the Y-axis do not exactly correspond in children:
image
image

File: docs/_includes/exampleControls.html [hunk 1]
```diff
@@ -14,6 +14,7 @@ <h5>Olli controls</h5>
     <li>y — jump to y-axis of current chart</li>
     <li>l — jump to legend of current chart</li>
     <li>t — jump to top level of current tree</li>
+    <li>Shift + Left / Right arrow — when chart has facets or layers, move to the corresponding item in the previous or next facet or layer</li>
     <!-- <li>W / A / S / D — when inside the grid of a scatterplot, move around the grid spatially</li> -->
```
Comment (ezralee) [inline docs/_includes/exampleControls.html:18]: Took a stab at wording... please fix :-)

File: packages/core/src/Runtime/OlliRuntime.ts [hunk 2]
```diff
@@ -167,6 +168,49 @@ export class OlliRuntime {
...
+
+  setFocusToLateralItem(currentItem: OlliRuntimeTreeItem, direction: 'left' | 'right') {
+    const { view, path } = this.getPathFromView(currentItem.olliNode)
+    const viewIndex = view.parent.children.indexOf(view);
+
+    const newViewIndex = direction === 'left' ? viewIndex - 1 : viewIndex + 1;
+    if (newViewIndex >= 0 && newViewIndex < view.parent.children.length) {
+      const newView = view.parent.children[newViewIndex];
+      const lateralNode = this.getNodeForPathFromView(newView, path);
+      const lateralItem = currentItem.tree.treeItems.find(ti => ti.olliNode === lateralNode);
```
Comment (ezralee) [inline packages/core/src/Runtime/OlliRuntime.ts:209]: should I try to use `olliNodeLookup` ?

Comment (jonathanzong) [reply packages/core/src/Runtime/OlliRuntime.ts:209]: nah this seems fine

**Pull request merged**